state/execution.go

```diff
@@ -423,18 +441,13 @@ func (blockExec *BlockExecutor) Commit(
 //---------------------------------------------------------
 // Helper functions for executing blocks and updating state
 
-func buildLastCommitInfo(block *types.Block, store Store, initialHeight int64) abci.CommitInfo {
+func BuildLastCommitInfo(block *types.Block, lastValSet *types.ValidatorSet, initialHeight int64) abci.CommitInfo {
```

Comment:
Can we structure this function at two levels, the outer being private, the inner being public, so that:

* The outer does the loading of the validators
* The inner does what is does in your patch

This way we avoid repeating the extracted code at all callsites.

Reply:
Yes, great idea!